package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,420 @@
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""H5MD trajectories --- :mod:`MDAnalysis.coordinates.H5MD`
+========================================================
+
+The `H5MD`_ file format is based upon `HDF5`_, which makes use of parallel
+file system features through the MPI-IO interface of the HDF5 library.
+The reader currently uses the `H5PY`_ library to access data from an H5MD file.
+
+.. _`H5MD`: https://nongnu.org/h5md/index.html
+.. _`HDF5`: https://www.hdfgroup.org/solutions/hdf5/
+.. _`H5PY`: http://docs.h5py.org/en/stable/
+
+
+Classes
+-------
+
+.. autoclass:: Timestep
+   :members:
+
+   .. attribute:: positions
+
+      coordinates of the atoms as a :class:`numpy.ndarray` of shape
+      `(n_atoms, 3)`
+
+   .. attribute:: velocities
+
+      velocities of the atoms as a :class:`numpy.ndarray` of shape
+      `(n_atoms, 3)`; only available if the trajectory contains velocities
+      or if the *velocities* = ``True`` keyword has been supplied.
+
+   .. attribute:: forces
+
+      forces of the atoms as a :class:`numpy.ndarray` of shape
+      `(n_atoms, 3)`; only available if the trajectory contains forces
+      or if the *forces* = ``True`` keyword has been supplied.
+
+
+.. autoclass:: H5MDReader
+   :members:
+
+"""
+
+import numpy as np
+import MDAnalysis as mda
+from MDAnalysis.coordinates import base, core
+try:
+    import h5py
+except ImportError:
+    HAS_H5PY = False
+else:
+    HAS_H5PY = True
+
+
+class Timestep(base.Timestep):
+    """H5MD Timestep
+    """
+    order = 'C'
+
+    def _init_unitcell(self):
+        return np.zeros((3, 3), dtype=np.float32)
+
+    @property
+    def dimensions(self):
+        """unitcell dimensions (*A*, *B*, *C*, *alpha*, *beta*, *gamma*)
+
+        lengths *A*, *B*, *C* are in the MDAnalysis length unit (Å), and
+        angles are in degrees.
+
+        Setting dimensions will populate the underlying native format
+        description (triclinic box vectors). If `edges
+        <https://nongnu.org/h5md/h5md.html#simulation-box>`_ is a matrix,
+        the box is of triclinic shape with the edge vectors given by
+        the rows of the matrix.
+        """
+        return core.triclinic_box(*self._unitcell)
+
+    @dimensions.setter
+    def dimensions(self, box):
+        self._unitcell[:] = core.triclinic_vectors(box)
+
+
+class H5MDReader(base.ReaderBase):
+    """Reader for the H5MD format.
+    (See `h5md documentation <https://nongnu.org/h5md/h5md.html#>`_
+    for a detailed overview)
+
+    Currently reads .h5md files with the following HDF5 hierarchy:
+
+    .. code-block:: text
+
+        Notation:
```

Comment:
DId you build the docs? Does this look ok?

Reply:
It looks something like this:

![ss](https://user-images.githubusercontent.com/59893899/88027429-16183800-caec-11ea-853c-a013dc88caf3.PNG)

Which I thought looked nice and clean, albeit kinda large. What do you think?

Reply:
That's fine.